File: Programming Basics Exam - 17 January 2016/CakeTycoon.js
function solve(input){
	
	var numberOfCakes = parseInt(input[0]);
	var flourNeeded = Number(input[1]);
	var flour = parseInt(input[2]);
	var truffles = parseInt(input[3]);
	var priceTruffle = parseInt(input[4]);

	var cakesToMake = flour / flourNeeded;

	if (Math.floor(cakesToMake) >= numberOfCakes)
	{
		var trufflesCost = priceTruffle * truffles;
		var cakePrice = ( trufflesCost / numberOfCakes) * 1.25;
		console.log("All products available, price of a cake: %s", cakePrice.toFixed(2));
	}
	else
	{
		var totalFlour = numberOfCakes * flourNeeded;
		var kilogramsNeeded = totalFlour - flour;
		console.log("Can make only %s cakes, need %s kg more flour", Math.floor(cakesToMake), kilogramsNeeded.toFixed(2)); 
	}
}

File: Programming Basics Exam - 17 January 2016/CakeTycoon.cs
using System;

namespace ProgrammingBasic
{
	class  CakeTycoon
	{
		static void Main(string[] args)
		{
			Int64 numberOfCakes = Int64.Parse(Console.ReadLine());
			double flourNeeded = double.Parse(Console.ReadLine());
			Int64 flour = Int64.Parse(Console.ReadLine());
			Int64 truffles = Int64.Parse(Console.ReadLine());
			Int64 priceTruffle = Int64.Parse(Console.ReadLine());

			double cakesToMake = flour / flourNeeded;

			if (Math.Floor(cakesToMake) >= numberOfCakes)
			{
				Int64 trufflesCost = priceTruffle * truffles;
				double cakePrice = ((double) trufflesCost / numberOfCakes) * 1.25;
				Console.WriteLine("All products available, price of a cake: {0:F2}", cakePrice);
			}
			else
			{
				double totalFlour = numberOfCakes * flourNeeded;
				double kilogramsNeeded = totalFlour - flour;
				Console.WriteLine("Can make only {0} cakes, need {1:F2} kg more flour", Math.Floor(cakesToMake), kilogramsNeeded); 
			}
		}
	}
}